lib/src/standardSubstitutions.cc

```diff
@@ -295,4 +295,39 @@ std::string StandardSubstitutions::initVariable(
     checkUnreplacedVariables(code, "initVar");
 
     return code;
-}
\ No newline at end of file
+}
+
+void StandardSubstitutions::currentSourceTimeCondition(
+    std::string &tcCode,
+    const CurrentSource *sc,
+    const DerivedParamNameIterCtx &scmDerivedParams,
+    const ExtraGlobalParamNameIterCtx &scmExtraGlobalParams)
+{
+    substitute(tcCode, "$(t)", "t");
+    value_substitutions(tcCode, sc->getCurrentSourceModel()->getParamNames(), sc->getParams());
+    value_substitutions(tcCode, scmDerivedParams.nameBegin, scmDerivedParams.nameEnd, sc->getDerivedParams());
+    name_substitutions(tcCode, "", scmExtraGlobalParams.nameBegin, scmExtraGlobalParams.nameEnd, sc->getName());
+    checkUnreplacedVariables(tcCode, sc->getName() + " : surrent source timeConditionCode");
+}
+
+void StandardSubstitutions::currentSourceInjection(
+    std::string &iCode,
+    const CurrentSource *sc,
+    const VarNameIterCtx &scmVars,
+    const DerivedParamNameIterCtx &scmDerivedParams,
+    const ExtraGlobalParamNameIterCtx &scmExtraGlobalParams,
+    const std::vector<FunctionTemplate> functions,
+    const std::string &ftype,
+    const std::string &rng)
+{
+    substitute(iCode, "$(t)", "t");
```

Comment:
It's always good to substitute in ``$(id)`` as the neuron ID too - if nothing else it lets you hack about indexing extra global variables :smile: